### PR TITLE
Fix to get data methods and pjax working together

### PR DESCRIPTION
**Pull request by derekisbusy**
I added this commit before #9432 but never got merged and ended up having to reset my branch because the branch was both ahead and behind in commits and wasn't sure how to resolve that with git. Anyway here are the updates applied again.. I also added support to fix @elastic-od issue in previous patch. I have tested this code and it does work. Please merge with master branch. Thanks!

### CODE REVIEW

File: framework/assets/yii.js [hunk 1]
```diff
@@ -143,16 +143,49 @@ yii = (function ($) {
...
          */
-        handleAction: function ($e) {
+        handleAction: function ($e,event) {
             var method = $e.data('method'),
                 $form = $e.closest('form'),
                 action = $e.attr('href'),
-                params = $e.data('params');
+                params = $e.data('params'),
+                pjax = $e.data('pjax'),
+                pjaxPushState = $e.data('pjax-push-state')? true : false,
```
Comment (samdark) [inline framework/assets/yii.js]: Casting to boolean could be done using `!!` i.e. `!!$e.data('pjax-push-state')`.

File: framework/assets/yii.js [hunk 2]
```diff
@@ -143,16 +143,49 @@ yii = (function ($) {
...
+                pjaxReplaceState = $e.data('pjax-replace-state')? true : false,
+                pjaxTimeout = $e.data('pjax-timeout')? true : false,
+                pjaxScrollTo = $e.data('pjax-scrollto')? true : false,
+                pjaxContainer,
+                pjaxOptions = {};
+            
+            if(pjax!==undefined && $.support.pjax) {
```
Comment (samdark) [inline framework/assets/yii.js]: Formatting seems to be off:

- Need spaces.
- Need `{` and `}`.

File: framework/widgets/Pjax.php [hunk 3]
```diff
@@ -109,7 +109,13 @@ public function init()
                 echo Html::tag('title', Html::encode($view->title));
             }
         } else {
-            echo Html::beginTag('div', $this->options);
+            echo Html::beginTag('div', array_merge([
+                'data-pjax-container'=>'',
+                'data-pjax-push-state'=>$this->enablePushState,
```
Comment (samdark) [inline framework/widgets/Pjax.php]: Missing spaces around `=>`

**Comment by derekisbusy** [comment]
@samdark File formatting has been updated

**Comment by SilverFire** [comment]
Looks good as for me

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2/commit/17e2255472a2b1a30fb1ce42a8937793214bbc46. Thanks! Please help identifying issues solved. There should be many...

**Comment by servocoder** [comment]
@derekisbusy, @samdark
Hey, I have found one more related issue.
Check my pull request please #10267
It is a small bugfix and I believe it should be included in the current milestone 2.0.7

**Pull request closed without merging**